### PR TITLE
chore: add perf test for select widget

### PR DESCRIPTION
**Pull request by prsidhu**
## Description

Add perf test for select widget.
Actions included:
1. rendering options list.
2. selecting an option

New script to run a single test - start-single-test.sh

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: test/select_perf 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.46 **(0)** | 37.91 **(-0.01)** | 36.15 **(0)** | 56.69 **(0.01)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0)** | 70.59 **(-1.96)** | 100 **(0)** | 93.33 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/perf/src/utils/utils.js [hunk 1]
```diff
@@ -104,8 +104,8 @@ exports.login = async (page) => {
   await page.waitForSelector(passwordSelector);
   await page.waitForSelector(buttonSelector);
 
-  await page.type(emailSelector, "hello@myemail.com");
-  await page.type(passwordSelector, "qwerty1234");
+  await page.type(emailSelector, "hello@mail.com");
```
Comment (SatishGandham) [inline app/client/perf/src/utils/utils.js:107]: You should not change these values.

File: app/client/perf/tests/select-widget.perf.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const { delay } = require("../src/utils/utils");
+process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;
+
+const SEL = {
+  select_button: ".select-button",
+  options_list: ".menu-virtual-list",
+  first_option_item: ".menu-item-text:nth-child(1)",
+};
+
+const actions = {
```
Comment (SatishGandham) [inline app/client/perf/tests/select-widget.perf.js:12]: These should be unique across the suite.
Move them to the actions file we have.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — DISMISSED** [review]

File: app/client/perf/start-single-test.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export PERF_TEST_ENV=dev
+cd tests
```
Comment (SatishGandham) [inline app/client/perf/start-single-test.sh:2]: What is this file for?

File: app/client/perf/src/utils/utils.js [hunk 4]
```diff
@@ -105,7 +105,7 @@ exports.login = async (page) => {
   await page.waitForSelector(passwordSelector);
   await page.waitForSelector(buttonSelector);
 
-  await page.type(emailSelector, "hello@myemail.com");
+  await page.type(emailSelector, "hello@ymail.com");
```
Comment (SatishGandham) [inline app/client/perf/src/utils/utils.js:108]: This email should not be changed

File: app/client/perf/tests/actions.js [hunk 5]
```diff
@@ -5,4 +5,6 @@ exports.actions = {
   UPDATE_POST_TITLE: "UPDATE_POST_TITLE",
   OPEN_MODAL: "OPEN_MODAL",
   CLOSE_MODAL: "CLOSE_MODAL",
+  OPEN_SELECT_MENU: "OPEN_SELECT_MENU",
+  CLICK_SELECT_OPTION_ITEM: "CLICK_SELECT_OPTION_ITEM",
 };
```
Comment (SatishGandham) [inline app/client/perf/tests/actions.js:10]: Please prefix your actions like this. If you have a better idea please share. Ideally earlier action should have been prefixed with golden.
SELECT_WIDGET_MENU_OPEN
SELECT_WIDGET_SELECT_OPTION

File: coverage-summary.json [hunk 6]
```diff
@@ -1,861 +1,1116 @@
...
-,"/Users/balajisoundararajan/projects/appsmith/app/client/src/utils/hooks/useIsScrolledToBottom.tsx": {"lines":{"total":15,"covered":15,"skipped":0,"pct":100},"functions":{"total":4,"covered":4,"skipped":0,"pct":100},"statements":{"total":21,"covered":20,"skipped":0,"pct":95.24},"branches":{"total":24,"covered":17,"skipped":0,"pct":70.83}}
-,"/Users/balajisoundararajan/projects/appsmith/app/client/src/utils/hooks/useOrg.tsx": {"lines":{"total":16,"covered":14,"skipped":0,"pct":87.5},"functions":{"total":2,"covered":2,"skipped":0,"pct":100},"statements":{"total":17,"covered":15,"skipped":0,"pct":88.24},"branches":{"total":11,"covered":3,"skipped":0,"pct":27.27}}
-,"/Users/balajisoundararajan/projects/appsmith/app/client/src/utils/hooks/usePositionedContainerZIndex.ts": {"lines":{"total":21,"covered":21,"skipped":0,"pct":100},"functions":{"total":4,"covered":4,"skipped":0,"pct":100},"statements":{"total":22,"covered":22,"skipped":0,"pct":100},"branches":{"total":14,"
```
Comment (SatishGandham) [inline coverage-summary.json:1]: Is this file supposed to be committed?

**Review by SatishGandham — CHANGES_REQUESTED** [review]

File: app/client/perf/start-single-test.sh (hunk 3 above)
Comment (prsidhu) [reply app/client/perf/start-single-test.sh:2]: simple script to execute a single test. I was having to set env repeatedly.

Comment (SatishGandham) [reply app/client/perf/start-single-test.sh:2]: This should not be part of GIT. Please remove it.
You can edit your bash profile to set the env variable.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**